tools/RAiDER/aria/prepFromGUNW.py

```diff
@@ -22,25 +24,24 @@
 DCT_POSTING = {'HRRR': 0.05, 'HRES': 0.10, 'GMAO': 0.10, 'ERA5': 0.10, 'ERA5T': 0.10}
 
 
-class GUNW(object):
-    def __init__(self, f:str, wm:str, out_dir:str):
-        self.path_gunw = f
-        self.wmodel    = wm
-        self.out_dir   = out_dir
-        self.spacing_m = int(DCT_POSTING[self.wmodel] * 1e5)
-
+@dataclass
+class GUNW:
+    path_gunw: str
+    wm: str
+    out_dir: str
 
-    def __call__(self):
+    def __post_init__(self):
         self.SNWE      = self.get_bbox()
         self.heights   = np.arange(-500, 9500, 500).tolist()
-        self.dates     = self.get_dates()
-        self.ref_time  = self.get_time()
-        self.look_dir  = self.get_look_dir()
+        self.dates, self.mid_time = self.get_datetimes()
+
+        self.look_dir   = self.get_look_dir()
         self.wavelength = self.get_wavelength()
-        self.name       = f'{self.dates[0]}-{self.dates[1]}'
-        self.OrbitFiles = self.get_orbit_files()
+        self.name       = self.make_fname()
```

Comment:
I assume this is only to make an intermediate file; does not propagate into the product?

Reply:
Right, this is the name of the .cfg file which is used by RAiDER